packages/web-app-files/src/composables/parentShare/useIncomingParentShare.ts

```diff
@@ -0,0 +1,34 @@
+import { buildShare } from '../../helpers/resources'
+import { useStore } from 'web-pkg/src/composables'
+import { computed, ref, unref } from '@vue/composition-api'
+import { useTask } from 'vue-concurrency'
+import { clientService } from 'web-pkg/src/services'
+
+export function useIncomingParentShare() {
+  const store = useStore()
+  const incomingParentShare = ref(null)
+  const sharesTree = computed(() => store.state.Files.sharesTree)
+
+  const loadIncomingParentShare = useTask(function* (signal, resource) {
+    let parentShare
+    for (const shares of Object.values(unref(sharesTree)) as any) {
```

Comment:
Please walk the sharesTree from the leaf up to the root in sorted manner to get the closest match in case of parent shares on multiple levels.

Reply:
Can we ensure that the `sharesTree` is already loaded?

Reply:
> Please walk the sharesTree from the leaf up to the root in sorted manner to get the closest match in case of parent shares on multiple levels.

It seems that this is already the case. Honestly, I don't know why, maybe the combination of `Object.values()` and `unref()` leads to this?

> Can we ensure that the sharesTree is already loaded?

Yes. `loadIncomingParentShare()` is being called in a `sharesTree` watcher in the `SharesPanel` component (= after the `sharesTree` has been loaded.

Reply:
👍